src/Infrastructure/Repository/TaskYamlRepository.php

```diff
@@ -433,4 +434,51 @@ protected function buildTaskSet(array $taskData, array $taskListData, array $tas
 
         return $task;
     }
+
+    /**
+     * @FIXME
+     *
+     * - This code is needed to be moved to separate repo (TaskPhpRepository)
+     * - To think about $this->existingTasks as well. I think we don't need them here.
```

Comment:
Sounds like a good idea to me. I think this topic should be discussed with @vol4onok

Reply:
Why? we've already discussed it. we need just rename TaskYaml repo and all

Reply:
@vol4onok  PHP tasks just seem to me like a separate concept and I think wrapping them into a repository will be more clear than passing as a simple iterable. This does not deny the idea that they should be part of file task repo. But on the other hand the situation gets more confusing because they are injected into persist repo too and used there separately from the file task repo. Maybe it will be clearer to inject the file repo into persist repo as a whole?

Reply:
I renamed the class, but non yaml tasks seem hidden and it's not clear what existingTasks are.